backend/src/test/java/com/crm/service/impl/CarServiceImplTest.java

```diff
@@ -63,4 +66,96 @@ public void getCarByIdShouldThrowExceptionForNotExistingEntry() {
         verify(carRepository, times(1)).findById(ONE);
         verify(carMapper, times(0)).convertToDto(any());
     }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinLengthLessThenSeventeenCharacters() {
+        //given
+        String VIN_TOO_SHORT = "V455";
+
+        //then
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_TOO_SHORT))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_INVALID_LENGTH);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinLengthMoreThenSeventeenCharacters() {
+        //given
+        String VIN_TOO_LONG = "V455467475455845FDLKPRTFGG";
+
+        //then
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_TOO_LONG))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_INVALID_LENGTH);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_O() {
+        //given
+        String VIN_WITH_CHAR_O = "VIN4003876543920O";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_O))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_I() {
+        //given
+        String VIN_WITH_CHAR_I = "VIN4003876543920i";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_I))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_Q() {
+        //given
+        String VIN_WITH_CHAR_Q = "VINq4003876543920";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_Q))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+
+    @Test
+    public void shouldGetCarByVin() {
+        //given
+        String vin = "VWVWVW12345678901";
+
+        //TODO
```

Comment:
TODO?
Not sure if these objects need to be specified - object parameters actually do not matter in this case. Mapping is not tested there, but only we need an object to verify if the right methods were invoked. Worth discussing :)

Reply:
Takie entity warto wyrzucić do jakiegoś Befora

Reply:
ale czy wrzucanie do `Before` ma jakiś sens już w momencie kiedy takie entity, póki co używamy tylko w tej konkretnej metodzie?

Reply:
Teraz w jednym ale w przyszłości pewnie będzie już w kilku. Warto od razu zadbać o ułatwienie pracy.

Reply:
Zrobione